grpc/src/test/java/com/linecorp/armeria/server/grpc/UnframedGrpcErrorHandlerUtilsTest.java

```diff
@@ -0,0 +1,145 @@
+                                           PreconditionFailure.Violation.newBuilder()
+                                                                        .setDescription("violation")
+                                                                        .build())
+                                   .build();
+        final BadRequest badRequest =
+                BadRequest.newBuilder()
+                          .addFieldViolations(
+                                  FieldViolation.newBuilder()
+                                                .setDescription("field violation")
+                                                .build())
+                          .build();
+        final RequestInfo requestInfo = RequestInfo.newBuilder()
+                                                   .setRequestId("requestid")
+                                                   .build();
+        final ResourceInfo resourceInfo = ResourceInfo.newBuilder()
+                                                      .setDescription("description")
+                                                      .build();
+        final Help help = Help.newBuilder()
+                              .addLinks(Link.newBuilder()
+                                            .setDescription("descrption")
+                                            .build())
+                              .build();
+        final LocalizedMessage localizedMessage = LocalizedMessage.newBuilder()
+                                                                  .setMessage("message")
+                                                                  .build();
+
+        final Status status = Status.newBuilder()
+                                    .setCode(
+                                              Code.UNKNOWN.getNumber())
+                                    .setMessage("Unknown Exceptions Test")
+                                    .addDetails(Any.pack(errorInfo))
+                                    .addDetails(Any.pack(retryInfo))
+                                    .addDetails(Any.pack(debugInfo))
+                                    .addDetails(Any.pack(quotaFailure))
+                                    .addDetails(Any.pack(preconditionFailure))
+                                    .addDetails(Any.pack(badRequest))
+                                    .addDetails(Any.pack(requestInfo))
+                                    .addDetails(Any.pack(resourceInfo))
+                                    .addDetails(Any.pack(help))
+                                    .addDetails(Any.pack(localizedMessage))
+                                    .build();
+
+        final JsonNode jsonNode = UnframedGrpcErrorHandlerUtils.convertErrorDetailToJsonNode(
+                status.getDetailsList(), UnframedGrpcErrorHandlerUtils.getErrorDetailsMarshaller());
+        final String expectedJsonString =
+                "[{\"@type\":\"type.googleapis.com/google.rpc.ErrorInfo\"," +
```

Comment:
Optional; It is hard to review this compact JSON. 😅 Should we add some indentation and readable newlines?